src/mail.sh

```diff
@@ -893,14 +962,19 @@ function parse_mail_options()
         shift
         ;;
       -v)
-        patch_version="$1$2"
+        options_values['PATCH_VERSION']="$1$2"
         shift 2
         ;;
       --)
         shift
         # if a reference is passed after the -- we need to account for it
-        [[ "$*" =~ -[[:digit:]]+ ]] && commit_ref=1
-        options_values['PASS_OPTION_TO_SEND_EMAIL']="$(str_strip "$* $patch_version")"
+        if [[ "$*" =~ -[[:digit:]]+ ]]; then
+          commit_count="${BASH_REMATCH[0]}"
+          options_values['REV_LIST']+="$commit_count "
+        fi
+        options_values['PASS_OPTION_TO_SEND_EMAIL']="$(str_strip "$* ${options_values['PATCH_VERSION']}")"
+        options_values['REV_LIST']+="$(check_revision "${options_values['PASS_OPTION_TO_SEND_EMAIL']}")"
+        rev_ret="$?"
```

Comment:
This is here because the auto generation of recipients depends on the range of commits the user intends to send, therefore a more robust parsing and explicit separation of said arguments was needed.